src/main/scala/com/fulcrumgenomics/bam/UpdateReadGroups.scala

```diff
@@ -0,0 +1,152 @@
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.fulcrumgenomics.bam
+
+import com.fulcrumgenomics.cmdline.{ClpGroups, FgBioTool}
+import com.fulcrumgenomics.util.ProgressLogger
+import dagr.commons.CommonsDef._
+import dagr.commons.io.Io
+import dagr.commons.util.LazyLogging
+import dagr.sopt._
+import htsjdk.samtools._
+import htsjdk.samtools.util.CloserUtil
+
+import scala.collection.JavaConversions._
+
+/**
+  * Updates one or more read groups.
+  *
+  * @author Nils Homer
+  */
+@clp(description =
+  """
+      |Updates one or more read groups and their identifiers.
+      |
+      |This tool will replace each read group with a new read group, including a new read group identifier.  If the read
+      |group identifier is not to be changed, it is recommended to use `samtools reheader` or Picard's
+      |`ReplaceSamHeader` instead as in this case only the header needs modification. If all read groups are to be
+      |assigned to one read group, it is recommended to use Picard's `AddOrReplaceReadGroups`.  Nonetheless, if the read
+      |group identifier also needs to be changed, use this tool.
+      |
+      |Each read group in the input file will be mapped to one and only one new read group identifier, unless
+      |`ignoreMissingReadGroups` is set.  A SAM header file should be given with the new read groups and the ID field
+      |foreach read group containing the new read group identifier.  An additional attribute ("FR") should be provided
+      |that gives the original read group identifier ("ID") to which this new read group corresponds.
+      |
+      |If `keepReadGroupAttributes` is true, then any read group attribute not replaced will be kept in the new read
+      |group.  Otherwise, only the attributes in the provided SAM header file will be used.
+    """,
+  group = ClpGroups.SamOrBam)
+class UpdateReadGroups
+(@arg(flag = "i", doc = "Input BAM file.") val input: PathToBam,
+ @arg(flag = "o", doc = "Output BAM file.") val output: PathToBam,
+ @arg(flag = "r", doc = "A SAM header file with the replacement read groups (see detailed usage).") val readGroupsFile: PathToBam,
+ @arg(flag = "k", doc = "Keep all read group attributes that are not replaced.") val keepReadGroupAttributes: Boolean = false,
+ @arg(flag = "g", doc = "If true ignore read groups not found in the replacement header, otherwise throw an error.") val ignoreMissingReadGroups: Boolean = false
```

Comment:
Does ignore mean a) don't futz with them in the header, or b) don't output reads from other read groups?